File: Source/ObjC/NSKVerticalImageLayout.m
//
//  NSKVerticalImageLayout.m
//  NSKButtonObjC
//
//  Created by NSSimpleApps on 29.08.16.
//  Copyright © 2016 NSSimpleApps. All rights reserved.
//

#import "NSKVerticalImageLayout.h"

@implementation NSKVerticalImageLayout

+ (CGSize)intrinsicContentSizeForDefault:(CGSize)defaultIntrinsicContentSize
                               imageRect:(CGRect)imageRect
                               titleRect:(CGRect)titleRect {
    
    CGSize intrinsicContentSize = defaultIntrinsicContentSize;
    intrinsicContentSize.height += MIN(imageRect.size.height, titleRect.size.height);
    
    return intrinsicContentSize;
}

@end

File: Source/Swift/NSKVerticalImageLayout.swift
//
//  NSKVerticalImageLayout.swift
//  NSKButtonSwift
//
//  Created by NSSimpleApps on 29.08.16.
//  Copyright © 2016 NSSimpleApps. All rights reserved.
//

import UIKit


internal class NSKVerticalImageLayout: NSKDefaultImageLayout {
    
    override static func intrinsicContentSize(forDefaultIntrinsicContentSize defaultIntrinsicContentSize: CGSize,
                                              imageRect: CGRect,
                                              titleRect: CGRect) -> CGSize {
        
        return CGSize(width: max(titleRect.width, imageRect.width),
                      height: titleRect.height + imageRect.height)
    }
}
